src/Compiler/Driver/CompilerOptions.fs

```diff
@@ -150,16 +152,16 @@ let PrintCompilerOption (CompilerOption (_s, _tag, _spec, _, help) as compilerOp
         | Some s -> s.Split [| ' ' |]
 
     let _finalColumn = Array.fold printWord flagWidth words
-    printfn "" (* newline *)
+    printer $"{nl}"
 
-let PrintPublicOptions (heading, opts) =
+let PrintPublicOptions (heading, opts) printer =
     if not (isNil opts) then
-        printfn ""
-        printfn ""
-        printfn "\t\t%s" heading
-        List.iter PrintCompilerOption opts
+        printer $"{nl}"
+        printer $"{nl}"
+        printer $"\t\t{heading}{nl}"
+        List.iter (PrintCompilerOption printer) opts
 
-let PrintCompilerOptionBlocks blocks =
+let PrintCompilerOptionBlocks blocks printer =
```

Comment:
I think better to take a TextWriter argument rather than a printer function.  That's more standard.

Reply:
Take it as the first argument.  Change `printfn` to `fprintfn` etc

Reply:
Also removed for good